barretenberg/cpp/src/barretenberg/stdlib/hash/poseidon2/poseidon2.test.cpp

```diff
@@ -0,0 +1,313 @@
+#include "poseidon2.hpp"
```

Comment:
mostly copied from pedersen.test.cpp, serves as consistency test with native poseidon2 implementation

Reply:
i think this is much better :)